src/components/App/App.js

```diff
@@ -42,6 +51,43 @@ class App extends Component {
     }
   }
 
+  componentDidMount() {
+    if (typeof window.ethereum !== 'undefined'){
+      window.ethereum.on('chainChanged', (chainId) => {
+        if (chainId !== MAINNET_CHAIN_ID_HEX) {
+          this.setState({ showNetworkChangeModal: true, chainId: parseChainIdHex(chainId)});
+        }
+      });
+
+      if(window.ethereum.isConnected()) {
+        window.ethereum.request({ method: 'eth_chainId' })
+          .then(chainId => {
+            if (chainId !== MAINNET_CHAIN_ID_HEX) {
+              this.setState({ showNetworkChangeModal: true, chainId: parseChainIdHex(chainId)});
```

Comment:
Same as above, but here instead of a listener the `setState` is happening after a promise resolution, which could happen by the time the component is unmounted, and result in an illegal `setState` call. You can always check if the component is still mounted and return early if it's not, or try to move the side effects outside of the component lifecycle (like within a thunk).

Again this is just a heads up because in practice in this dApp this component will never unmount, but is something that could happen later if the dapp grows into more pages.